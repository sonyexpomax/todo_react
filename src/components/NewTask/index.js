import {addTaskAction} from '../../redux/actions/tasks';
import { connect } from 'react-redux';
import NewTask from './NewTask';
import PropTypes from 'prop-types';
import React from 'react';

const NewTaskContainer = (props) => <NewTask {...props} />;

NewTaskContainer.propTypes = {
    listId: PropTypes.number,
    createTask: PropTypes.func
};

function mapDispatchToProps (dispatch) {
    return ({
        createTask: (content, listId) => dispatch(addTaskAction(content, listId))
    });
}

export default connect(null, mapDispatchToProps)(NewTaskContainer)