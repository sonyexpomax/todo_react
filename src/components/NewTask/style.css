.td-new-task-wrap{
    padding:0 20px;

}
.td-new-task-wrap h3{
    color: #4f100f;
    padding: 0;
    margin: 0;
}
.td-new-task-add-text{
    width: 87%;
    font-family: "Source Code Pro", Monaco, monospace;
    font-size: 16px;
    margin: 5px 0;
    border: 1px solid #5d5d5d;
    border-right: none;
    padding: 5px 1.5%;
}
.td-new-task-add-btn{
    width: 9%;
    margin:10px 0 ;
    background-color: #3b8e00;
    color: #ffffff;
    border: 1px solid #5d5d5d;
    font-family: "Source Code Pro", Monaco, monospace;
    font-size: 16px;
    padding: 5px 0;
}
.td-new-task-add-btn:hover{
    background-color: #215000;
    cursor: pointer;
}
.td-new-task-add-form{
    margin: 0 10px 0 30px;
}