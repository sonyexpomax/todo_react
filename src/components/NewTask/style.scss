@import '../../utils/vars';
@import '../../utils/templates';

.td-new-task-wrap{
    padding:0;
    margin: 0;
    @extend %input-text;

    input{
        border-top: none;
        margin: 0;
    }
}
.td-new-task-btn{
    display: none;
    @extend %list-item;
    padding:0 1.4%;
    margin: 0;
    border-top: none;
}
.td-new-task-add-btn{
    @extend %confirm-btn;
    display: inline-block;
    background-color: $color-green;
}
.td-new-task-cancel-btn{
    @extend %cancel-btn;
    display: inline-block;
}
.td-new-task-visible-btn{
    display: inline-block !important;
}


//.td-new-task-wrap h3{
//    color: #4f100f;
//    padding: 0;
//    margin: 0;
//}
//.td-new-task-add-text{
//    width: 87%;
//    font-family: "Source Code Pro", Monaco, monospace;
//    font-size: 16px;
//    margin: 5px 0;
//
//    padding: 5px 1.5%;
//}
//.td-new-task-add-btn{
//    width: 9%;
//    margin:10px 0 ;
//    background-color: #3b8e00;
//    color: #ffffff;
//    border: 1px solid #5d5d5d;
//    font-family: "Source Code Pro", Monaco, monospace;
//    font-size: 16px;
//    padding: 5px 0;
//}
//.td-new-task-add-btn:hover{
//    background-color: #215000;
//    cursor: pointer;
//}
