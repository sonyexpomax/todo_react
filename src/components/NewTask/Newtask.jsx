import React, { Component } from "react";
import { addTaskAction } from '../../redux/actions/tasks';
import './style.css';

class NewTask extends Component{

    constructor(props){
        super(props);
        this.state = {
            newTask: '',
        };
    }

    onChangeName = (e) => {
        this.setState({newTask: e.target.value});
    };

    onSubmit = (e) => {
        e.preventDefault();
        const { newTask } = this.state;
        const { dispatch } = this.props;
        if (newTask) {
            dispatch(addTaskAction(this.props.listId, newTask));
        }
    };

    render(){
        return (
            <div className={'new-task'}>
                <form onSubmit={this.onSubmit} className={'add-form'}>
                    <input onChange={this.onChangeName} placeholder="Add new task in this list" className={'add-task-text'} />
                    <button type="submit" className={'add-task-btn'}>Add</button>
                </form>
            </div>
        );
    }
}

export default NewTask;


