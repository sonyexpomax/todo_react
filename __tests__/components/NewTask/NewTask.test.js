import NewTask from "../../../src/components/NewTask/NewTask.jsx";
import React from 'react';
import { shallow, mount, render } from 'enzyme';

describe('Component NewTask', () => {
    const props = {
      listId: 1
    };
    it('start render component', () => {
        const component = shallow(<NewTask {...props}/>);
        expect(component).toMatchSnapshot();
    });


    it('click add task with empty input field', () => {
        const component = shallow(<NewTask {...props}/>);
        const text = component.find('input').text();
        expect(text).toEqual('');
    });
});


// import React from 'react';
// import NewTask from "../../../src/components/NewTask/NewTask";
// import renderer from 'react-test-renderer';
//
// test('Link changes the class when hovered', () => {
//
//     const props = {
//         listId: 1
//     };
//
//     const component = renderer.create(
//         <NewTask {...props}/>,
//     );
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//
//     // // manually trigger the callback
//     // tree.props.onChangeName();
//     // // re-rendering
//     // tree = component.toJSON();
//     // expect(tree).toMatchSnapshot();
//     //
//     // // manually trigger the callback
//     // tree.props.onSubmit();
//     // // re-rendering
//     // tree = component.toJSON();
//     // expect(tree).toMatchSnapshot();
// });