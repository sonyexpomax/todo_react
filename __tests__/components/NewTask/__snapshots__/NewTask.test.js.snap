// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component NewTask start render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewTask
    listId={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="td-new-task-add-form"
        onSubmit={[Function]}
      >
        <input
          className="td-new-task-add-text"
          onChange={[Function]}
          placeholder="Add new task in this list"
          value=""
        />
        <button
          className="td-new-task-add-btn"
          type="submit"
        >
          Add
        </button>
      </form>,
      "className": "td-new-task-wrap",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="td-new-task-add-text"
            onChange={[Function]}
            placeholder="Add new task in this list"
            value=""
          />,
          <button
            className="td-new-task-add-btn"
            type="submit"
          >
            Add
          </button>,
        ],
        "className": "td-new-task-add-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "td-new-task-add-text",
            "onChange": [Function],
            "placeholder": "Add new task in this list",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "className": "td-new-task-add-btn",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="td-new-task-add-form"
          onSubmit={[Function]}
        >
          <input
            className="td-new-task-add-text"
            onChange={[Function]}
            placeholder="Add new task in this list"
            value=""
          />
          <button
            className="td-new-task-add-btn"
            type="submit"
          >
            Add
          </button>
        </form>,
        "className": "td-new-task-wrap",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="td-new-task-add-text"
              onChange={[Function]}
              placeholder="Add new task in this list"
              value=""
            />,
            <button
              className="td-new-task-add-btn"
              type="submit"
            >
              Add
            </button>,
          ],
          "className": "td-new-task-add-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "td-new-task-add-text",
              "onChange": [Function],
              "placeholder": "Add new task in this list",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add",
              "className": "td-new-task-add-btn",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
